### PR TITLE
[ENH] Refactor `write_modality_agnostic_files` and add unit tests

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Currently, the bids version of `dataset_description.json` is hardcoded deep in a private function of the `iotools` module:

https://github.com/aramis-lab/clinica/blob/70bac93cb7bc62fd62ba6dd4685c1258928df314/clinica/iotools/bids_utils.py#L440-L441

It is also out of date and should comply (I think...?) with the version defined here:

https://github.com/aramis-lab/clinica/blob/70bac93cb7bc62fd62ba6dd4685c1258928df314/clinica/iotools/bids_dataset_description.py#L8

This PR proposes to:

- Use `BIDS_VERSION` value as the default value
- Allow `write_modality_agnostic_files` to write another version if needed
- Refactor this function into smaller private functions handling only one file each
- Add unit tests for all these functions

### CODE REVIEW

**Comment by ghisvail** [comment]
> It is also out of date and should comply (I think...?)

Until [this](https://github.com/aramis-lab/clinica/pull/593) is merged, the hardcoded version is the correct one.

**Comment by ghisvail** [comment]
Otherwise, this is a welcome refactoring 👍

**Comment by NicolasGensollen** [comment]
>Until https://github.com/aramis-lab/clinica/pull/593 is merged, the hardcoded version is the correct one.

Right, didn't realize you updated this in #593, thanks for noticing.
Let's wait for #593 to be merged and I'll rebase to get the refactoring + tests

**Comment by ghisvail** [comment]
@NicolasGensollen You can resume this if you like. I fixed the conflicts introduced by my recent PRs.

**Review by omar-rifai — APPROVED** [review]
Hello @NicolasGensollen, looks good, thanks!  Can you please rebase and I'll merge if the tests succeed.

**Review by omar-rifai — APPROVED** [review]

File: clinica/iotools/bids_utils.py
Comment (ghisvail) [inline clinica/iotools/bids_utils.py:474]: Let me suggest to use secure URLs when possible.

```suggestion
            f"More information on https://www.clinica.run\n"
```

File: clinica/iotools/bids_utils.py
Comment (ghisvail) [inline clinica/iotools/bids_utils.py:490]: We are now compliant with BIDS for PET.

```suggestion
        f.write("\n".join(["conversion_info/"]))
```

**Review by ghisvail — APPROVED** [review]

**Pull request merged**
